packages/redux/src/store.ts

```diff
@@ -55,8 +55,13 @@ export interface WorkspaceSettings {
     windows?: { [id: string]: ValidKeyState };
     mac?: { [id: string]: ValidKeyState };
   };
-  defaultNotebookSettings: {
+  notebookSettings: {
     isMinimapEnabled?: boolean;
+    formatOnSave?: boolean;
+    ruffSettings?: {
```

Comment:
Nest this under `python` namespace, e.g.

```
notebookSettings: {
  ...
  python: {
    linter: {
      isEnabled?: boolean;
      config: ...,
    };
  }
}
```

Defined on the server, namespaced under `web.user` or `web.client` (which they aren't right now), and we should automatically load all of those set values and transfer them (rather than having to explicitly `client.configuration.list`):
```
web.client.notebookSettings.isMinimapEnabled
web.client.notebookSettings.python.linter.isEnabled
```
Then do a deep merge

Reply:
Added these under `web.user` and tested locally by adding `web.user.notebookSettings.python.linter.isEnabled` to my `dh-defaults.prop#client.configuration.list`. Then set the prop in my local prop file and tested in incognito that the server setting default was applied